ethers-ext/src/core/util.ts

```diff
@@ -43,3 +43,175 @@ export const HexStr = {
     return ethers.utils.hexZeroPad( value, length);
   }
 };
+
+// Klaytn Type Enumeration
+export enum Klaytn {
+  // Basic
+  TxTypeValueTransfer = 0x08,
+  TxTypeValueTransferMemo = 0x10, 
+  TxTypeAccountUpdate = 0x20, 
+  TxTypeSmartContractDeploy = 0x28, 
+  TxTypeSmartContractExecution = 0x30, 
+  TxTypeCancel = 0x38, 
+
+  // Fee Delegation
+  TxTypeFeeDelegatedValueTransfer = 0x09,
+  TxTypeFeeDelegatedValueTransferMemo = 0x11, 
+  TxTypeFeeDelegatedAccountUpdate = 0x21, 
+  TxTypeFeeDelegatedSmartContractDeploy = 0x29, 
+  TxTypeFeeDelegatedSmartContractExecution = 0x31, 
+  TxTypeFeeDelegatedCancel = 0x39,
+
+  // Partial Fee Delegation 
+  TxTypeFeeDelegatedValueTransferWithRatio = 0x0a,
+  TxTypeFeeDelegatedValueTransferMemoWithRatio = 0x12,
+  TxTypeFeeDelegatedAccountUpdateWithRatio = 0x22, 
+  TxTypeFeeDelegatedSmartContractDeployWithRatio = 0x2a,
+  TxTypeFeeDelegatedSmartContractExecutionWithRatio = 0x32,
+  TxTypeFeeDelegatedCancelWithRatio = 0x3a,
+
+  // Account Key Type
+  AccountKeyLegacy = 0x01,
+  AccountKeyPublic = 0x02,
+  AccountKeyFail = 0x03, 
+  AccountKeyWeightedMultiSig = 0x04, 
+  AccountKeyRoleBased = 0x05
+};
+
+
+// For Klay unit
+// https://github.com/ethers-io/ethers.js/blob/main/src.ts/utils/units.ts 
+// https://github.com/klaytn/caver-js/blob/82ba48c52bd3fc2738ccba40f67a4bd0f206822a/packages/caver-utils/src/index.js#L241C1-L420C2
+
+const names = [
+  "peb",
+  "kpeb",
+  "Mpeb",
+  "Gpeb",
+  "ston",
+  "Ston",
+  "uKLAY",
+  "mKLAY",
+  "KLAY",
+  "kKLAY",
+  "MKLAY",
+  "GKLAY",
+  "TKLAY"
+];
+
+const KlayUnit = [
+  { unit: 'peb', pebFactor: 0 },
+  { unit: 'kpeb', pebFactor: 3 },
+  { unit: 'Mpeb', pebFactor: 6 },
+  { unit: 'Gpeb', pebFactor: 9 },
+  { unit: 'ston', pebFactor: 9 },
+  { unit: 'Ston', pebFactor: 9 },
+  { unit: 'uKLAY', pebFactor: 12 },
+  { unit: 'mKLAY', pebFactor: 15 },
+  { unit: 'KLAY', pebFactor: 18 },
+  { unit: 'kKLAY', pebFactor: 21 },
+  { unit: 'MKLAY', pebFactor: 24 },
+  { unit: 'GKLAY', pebFactor: 27 },
+  { unit: 'TKLAY', pebFactor: 30 },
+];
+
+function getFactor(unit: string): number {
+  for ( let i=0; i < KlayUnit.length ; i++ ) {
+    if ( KlayUnit[i].unit === unit) {
```

Comment:
toLowerCase to both sides please.

Reply:
But LowerCase comparison arises some problem. 
In Klaytn specification, we can not distinguish between mKLAY and MKLAY. 
https://docs.klaytn.foundation/content/klaytn/design/klaytn-native-coin-klay

Reply:
I will try to add toLowerCase to both sides + comparing 1st characters.

Reply:
I see the ambiguity. Thanks.